controller/workitem_whitebox_test.go

```diff
@@ -329,6 +334,112 @@ func (rest *TestWorkItemREST) TestConvertWorkItems() {
 	})
 }
 
+func (rest *TestWorkItemREST) TestConvertWorkItemsToCSV() {
+	rest.T().Run("ok - result set", func(t *testing.T) {
+		// given
+		fxt := tf.NewTestFixture(t, rest.DB, tf.CreateWorkItemEnvironment(),
+			tf.Spaces(1, func(fxt *tf.TestFixture, idx int) error {
+				fxt.Spaces[0].SpaceTemplateID = spacetemplate.SystemAgileTemplateID
+				return nil
+			}),
+			tf.Labels(3, tf.SetLabelNames("important", "backend", "ui")),
+			tf.WorkItems(3, func(fxt *tf.TestFixture, idx int) error {
+				wi := fxt.WorkItems[idx]
+				if idx < 2 {
+					wi.Type = uuid.FromStringOrNil("2853459d-60ef-4fbe-aaf4-eccb9f554b34") // Task
+					wi.Fields[workitem.SystemLabels] = []string{fxt.LabelByName("important").ID.String(), fxt.LabelByName("backend").ID.String()}
+					wi.Fields[workitem.SystemState] = interface{}("New")
+					wi.Fields["effort"] = 42.0 // Effort is in Task and Defect, will go in same column
+				} else {
+					wi.Type = uuid.FromStringOrNil("fce0921f-ea70-4513-bb91-31d3aa8017f1") // Defect
+					wi.Fields[workitem.SystemLabels] = []string{fxt.LabelByName("ui").ID.String()}
+					wi.Fields[workitem.SystemState] = interface{}("New")
+					wi.Fields["effort"] = 23.0
+					wi.Fields["severity"] = interface{}("SEV1 - Urgent") // default is SEV3
+				}
+				return nil
+			}),
+		)
+		wis := []workitem.WorkItem{*fxt.WorkItems[0], *fxt.WorkItems[1], *fxt.WorkItems[2]}
+		wits, err := loadWorkItemTypesFromPtrArr(rest.Ctx, rest.GormDB, fxt.WorkItems)
+		require.NoError(t, err)
+		// when
+		convertedWIs, err := ConvertWorkItemsToCSV(rest.Ctx, rest.GormDB, wits, wis)
+		require.NoError(t, err)
+		// parse the resulting CSV
+		var entities []map[string]string
+		csvReader := csv.NewReader(bytes.NewBufferString(convertedWIs))
+		// read header line
+		parsedHeaderKeys, err := csvReader.Read()
+		require.NoError(t, err)
+		// read the entities from the CSV
+		for {
+			line, err := csvReader.Read()
+			if err == io.EOF {
+				break
+			} else {
+				require.NoError(t, err)
+			}
+			entity := make(map[string]string)
+			for idx := range line {
+				entity[parsedHeaderKeys[idx]] = line[idx]
+			}
+			entities = append(entities, entity)
+		}
+		require.Len(t, entities, 3)
+		// now run tests on the maps.
+		// check a common field available thru the base type.
```

Comment:
~thru~ => through

Reply:
I guess that is debatable as `thru` is American English, no?

Reply:
I don't know. If there is a word like `thru` we should keep it :)

Reply:
Fixed 7ae9140.